src/mesh/mesh.cxx

```diff
@@ -58,11 +58,14 @@ int Mesh::get(std::string &sval, const std::string &name) {
 }
 
 /// Get an integer
-int Mesh::get(int &ival, const std::string &name) {
+int Mesh::get(int &ival, const std::string &name, int def) {
   TRACE("Mesh::get(ival, %s)", name.c_str());
 
-  if (source == nullptr or !source->get(this, ival, name))
+  if (source == nullptr or !source->get(this, ival, name, def)) {
+    // set ival to default in source==nullptr too:
+    ival = def;
```

Comment:
This is ugly: if `source == nullptr` then the default doesn't get set by `get`... so we set it silently here.  Without this, four of the integrated tests fail.

Reply:
Could split it into two separate conditionals:

```cpp
if (source == nullptr) {
   output_warn << "No source; setting default"; // but better, obvs
   ival = def;
   return 1;
}
return source->get(this, ival, name, def);
```

Reply:
`GridDataSource::get` returns `true` on success, whereas `Mesh::get` returns false. So this just needed `return !source->get(...);`